crates/tx5-connection/src/lib.rs

```diff
@@ -105,6 +105,40 @@ impl<T: 'static + Send> CloseSend<T> {
             Err(ErrorKind::BrokenPipe.into())
         }
     }
+
+    pub async fn send(&self, t: T) -> Result<()> {
+        let res = tokio::time::timeout(
+            // hard-coded to 1 minute for now. This indicates a system
+            // is very backed up, and is here just to prevent forever hangs
+            std::time::Duration::from_secs(60),
```

Comment:
lemme know if y'all are okay with this as an outside safeguard, or if it needs to be configurable